src/backend/columnar/columnar_customscan.c

```diff
@@ -670,7 +683,20 @@ ExprReferencesRelid(Expr *expr, Index relid)
 
 
 /*
- * CheckPushdownClause tests to see if clause is a candidate for pushing down
+ * DeparseExprForRelation deparses node for given relation.
+ */
+static char *
+DeparseExprForRelation(Node *node, Oid relationId)
+{
+	bool useTableNamePrefix = true;
+	bool showImplicitCast = false;
+	List *context = deparse_context_for(get_rel_name(relationId), relationId);
```

Comment:
I was hoping to get aliasname from somewhere and using it instead of relname but couldn't spend much time on this.
Since it is mostly for debugging purposes, I thought that we might be okay with that.

Reply:
I think you're right.
I've run all the regression tests by setting "columnar.planner_debug_level=notice", and got this error from one of the tests:

```diff
 explain (costs off, summary off) select * from
   INT8_TBL_columnar a left join lateral
   (select b.q1 as bq1, c.q1 as cq1, least(a.q1,b.q1,c.q1) from
    INT8_TBL_columnar b cross join INT8_TBL_columnar c) ss
   on a.q2 = ss.bq1;
-                          QUERY PLAN
----------------------------------------------------------------------
- Nested Loop Left Join
-   ->  Custom Scan (ColumnarScan) on int8_tbl_columnar a
-         Columnar Projected Columns: q1, q2
-   ->  Nested Loop
-         ->  Custom Scan (ColumnarScan) on int8_tbl_columnar c
-               Columnar Projected Columns: q1
-         ->  Custom Scan (ColumnarScan) on int8_tbl_columnar b
-               Filter: (a.q2 = q1)
-               Columnar Projected Columns: q1
-               Columnar Chunk Group Filters: (a.q2 = q1)
-(10 rows)
-
+NOTICE:  columnar planner: adding CustomScan path for a
+DETAIL:  unparameterized; 0 clauses pushed down
+NOTICE:  columnar planner: adding CustomScan path for a
+DETAIL:  parameterized by rels {b}; 0 clauses pushed down
+NOTICE:  columnar planner: adding CustomScan path for a
+DETAIL:  parameterized by rels {b, c}; 0 clauses pushed down
+NOTICE:  columnar planner: adding CustomScan path for a
+DETAIL:  parameterized by rels {c}; 0 clauses pushed down
+NOTICE:  columnar planner: adding CustomScan path for b
+DETAIL:  parameterized by rels {a}; 1 clauses pushed down
+NOTICE:  columnar planner: adding CustomScan path for b
+DETAIL:  parameterized by rels {a, c}; 1 clauses pushed down
+ERROR:  bogus varno: 4
```

Reply:
I think I should definitely revert the changes that I made to log quals

Reply:
(reverted)